Realm/ObjectServerTests/SwiftObjectServerTests.swift

```diff
@@ -3079,6 +2223,124 @@ class AsyncAwaitObjectServerTests: SwiftSyncTestCase {
             }
         }
     }
+
+    func testCallResetPasswordAsyncAwait() async throws {
+        let email = "realm_tests_do_autoverify\(randomString(7))@\(randomString(7)).com"
+        let password = randomString(10)
+        try await app.emailPasswordAuth.registerUser(email: email, password: password)
+        do {
+            try await app.emailPasswordAuth.callResetPasswordFunction(email: email,
+                                                                      password: randomString(10),
+                                                                      args: [[:]])
+        } catch {
+            XCTAssertNotNil(error)
+        }
+    }
+
+    func testAppLinkUserAsyncAwait() async throws {
+        let email = "realm_tests_do_autoverify\(randomString(7))@\(randomString(7)).com"
+        let password = randomString(10)
+        try await app.emailPasswordAuth.registerUser(email: email, password: password)
+
+        let syncUser = try await self.app.login(credentials: Credentials.anonymous)
+        XCTAssertNotNil(syncUser)
+
+        let credentials = Credentials.emailPassword(email: email, password: password)
+        let linkedUser = try await syncUser.linkUser(credentials: credentials)
+        XCTAssertNotNil(linkedUser)
+        XCTAssertEqual(linkedUser.id, app.currentUser?.id)
+        XCTAssertEqual(linkedUser.identities.count, 2)
+    }
+
+    // MARK: - Objective-C async await
+    func testPushRegistrationAsyncAwait() async throws {
+        let email = "realm_tests_do_autoverify\(randomString(7))@\(randomString(7)).com"
+        let password = randomString(10)
+        try await app.emailPasswordAuth.registerUser(email: email, password: password)
+
+        _ = try await app.login(credentials: Credentials.emailPassword(email: email, password: password))
+
+        let client = app.pushClient(serviceName: "gcm")
+        try await client.registerDevice(token: "some-token", user: app.currentUser!)
+        try await client.deregisterDevice(user: app.currentUser!)
+        XCTAssertTrue(true)
+    }
+
+    func testEmailPasswordProviderClientAsyncAwait() async throws {
+        let email = "realm_tests_do_autoverify\(randomString(7))@\(randomString(7)).com"
+        let password = randomString(10)
+        try await app.emailPasswordAuth.registerUser(email: email, password: password)
+
+        do {
+            try await app.emailPasswordAuth.confirmUser("atoken", tokenId: "atokenid")
+        } catch {
+            XCTAssertNotNil(error)
```

Comment:
Are these trying to verify that the call does throw? If so there needs to be a XCTFail() in the do block after the thing that should fail.

Reply:
same as the above comment